### DISCUSSION TITLE
Is there a way to require manual confirmation?

### DISCUSSION THREAD

**Discussion by dragdragdrag** [Flexget/Flexget#3913] [Q&A]
I have a `season_premiere` task. Usually I like what it does, but no matter how hard I try I can't always get everything I'm not interested in filtered out. I think often the metadata I want to filter on just doesn't exist yet when these obscure shows are so fresh.

I initially wanted a way to delay judgement on a candidate if particular metadata can't be found. In particular, if `tvdb_genres` or `tvdb_network` is empty I'd like it to be put in quarantine until some metadata exists for that, then judged. Maybe if a long enough period goes by and there's still no metadata it'd be rejected. But I couldn't figure out a way to do that. (Did I miss something?)

Assuming the above is impossible, perhaps it could ask for a manual confirmation on each one.

The way I imagine it working is that it'd go through the existing filters, and if wasn't rejected it'd go into a list of things requiring manual confirmation. I'd manually say yes or no to each one. In the simple case that'd work by running a CLI command to see the queue of things for confirmation, and running another CLI command for each acceptance/rejection. Or in the more sophisticated case it'd be nice if it'd message me on Telegram and allow me to respond to it there.

Is any of this currently possible?

**Comment by gazpachoking** [comment]
You can use the [require_field](https://flexget.com/en/Plugins/require_field) plugin to reject everything that hasn't gotten some metadata filled.
For manual confirmation [pending_list](https://flexget.com/Plugins/List/pending_list) is the right plugin. You add to that list, then you can emit only the ones that have been approved via the (webui? or) cli. You can certainly have it notify you (on telegram or otherwise) if something is added to this list, I'm less sure if we have the parts to approve via telegram. (maybe something fancy with from_telegram plugin?)

**Comment by dragdragdrag** [comment]
> You can use the [require_field](https://flexget.com/en/Plugins/require_field) plugin to reject everything that hasn't gotten some metadata filled.

Hm, I don't really want to reject it outright though. I'd ideally want it to try again some time later in the hope that metadata now exists.

> For manual confirmation [pending_list](https://flexget.com/Plugins/List/pending_list) is the right plugin.

Ah, great. I'll look at that. Thanks.

**Reply by gazpachoking** [reply]
> Hm, I don't really want to reject it outright though. I'd ideally want it to try again some time later in the hope that metadata now exists.

Ahh. You can manually use backlog plugin to make sure an entry sticks around, even if it goes away from your source. e.g. `backlog: 5 days` Now every entry that arrives on the input plugin will keep getting considered for however many days you specified in the backlog plugin. If within that time metadata shows up it'll be accepted.

**Reply by dragdragdrag** [reply]
Oh, that sounds perfect.

But I can't find documentation. The link "backlog plugin" from https://flexget.com/en/CLI/backlog is broken, similarly the link to it at https://flexget.com/CLI

I expected to find something at https://flexget.com/en/Plugins but "backlog" doesn't appear on the page.

Can you help me configure this? I'm starting with something like

```yaml
templates:
  premieres:
    series_premiere:
      allow_teasers: no
      <<: *series_settings
    require_field:
      - tvdb_genres
      - tvdb_network
    regexp:
      reject:
        - "reality": {from: tvdb_genres}
        - "soap": {from: tvdb_genres}
        - "^Network 1|Other Network$": {from: tvdb_network}
```

which is put into a task along with the input plugin config (which I can show if necessary).

I'll try just adding `backlog: 5 days` and hope that does what I want, but if you see this and you know there's more to it, please let me know.

**Reply by dragdragdrag** [reply]
My immediate concern is that if they're currently being accepted even when I don't want them to be, surely nothing would be stopping them still. It seems like `require_field` isn't doing what it's meant to?

**Reply by dragdragdrag** [reply]
Actually, to stop a lot of back and forth, I wonder if you'd be willing to draw up an example task which achieves this?

- Get input from a feed or whatever
- Use the `series_premiere` plugin
- Backlog it if `tvdb_genres` or `tvdb_network` are unknown or empty
- Reject it if some regexes match based on those fields
- Reject it if the backlog times out (or is this implied)
- Accept it only if those fields were present and the regexes didn't reject it
- Things which are on the backlog try again occasionally (presumably next time the task runs?)

**Reply by gazpachoking** [reply]
> My immediate concern is that if they're currently being accepted even when I don't want them to be, surely nothing would be stopping them still. It seems like require_field isn't doing what it's meant to?

Yeah, that sounds like the main issue. You are sure it actually isn't getting filled for the stuff you think should be rejected? Maybe try some tests with --dump so that you can see all the fields.

Once that bit is sorted I think you have everything else in place.

**Reply by dragdragdrag** [reply]
I added some fields to my notifier output and waited.

I just had one come in which says this:

```
...
tvdb_genres: ['Comedy', 'Crime']
tvdb_network: 
tvdb_first_air_date: 2024-01-11 00:00:00
```

Why would that be allowed through, given that I have
```
    require_field:
      - tvdb_genres
      - tvdb_network
```

**Reply by gazpachoking** [reply]
Ahh. It seems that the lookup succeeds, and tvdb_network is getting filled in as an empty string. I'm not 100% sure if the behavior of require_field should change here or not. It will only reject if the field doesn't exist, or is `None`. (There might be an argument that tvdb lookup should fill that field as `None` too, rather than an empty string.

You could work around this with the `if` plugin, given that I'm not exactly sure where the fix should go.
```yaml
if:
- not tvdb_network|d(None): reject
```
The `d(None)` isn't required, but prevents an error message if the tvdb_network field doesn't exist by using `None` as the default in that case.
This will work because `None` and the empty string are both falsy in python (and jinja) so either one will trigger the rejection.

**Reply by dragdragdrag** [reply]
OK, I'll give this a try. Thanks again for your help.

I wonder if you could help me with an example of how I'd get the entries caught by either `require_field` or `if -> not tvdb_network|d(None)` to be backlogged, in case that data is filled in upstream in the following week or whatever and so it can make a more informed decision? Or is that not possible?

**Reply by gazpachoking** [reply]
> I wonder if you could help me with an example of how I'd get the entries caught by either require_field or if -> not tvdb_network|d(None) to be backlogged, in case that data is filled in upstream in the following week or whatever and so it can make a more informed decision? Or is that not possible?

If you manually enable backlog plugin it will store every entry from the task in the backlog. You shouldn't need to do anything special for that.
